File: csharpcore-Verify.xunit/GildedRoseFinal/StandardItem.cs
namespace GildedRoseKata;

public class StandardItem : Item
{
    private const int DAILY_QUALITY_DECREASE_BEFORE_EXPIRY = 1;
    private const int DAILY_QUALITY_DECREASE_AFTER_EXPIRY = 2;

    public StandardItem(string name, int sellInDays, int quality) : base(name, sellInDays, quality)
    {
    }

    public override void UpdateQuality()
    {
        if (IsExpired())
        {
            DecreaseQuality(DAILY_QUALITY_DECREASE_AFTER_EXPIRY);
        }
        else
        {
            DecreaseQuality(DAILY_QUALITY_DECREASE_BEFORE_EXPIRY);
        }
    }
}

File: csharpcore-Verify.xunit/GildedRoseFinal/StandardItem.java
package gildedrose.item;

import gildedrose.Item;

public class StandardItem extends Item{
	private static final int DAILY_QUALITY_DEGREDATION_BEFORE_EXPIRY = 1;
	private static final int DAILY_QUALITY_DEGREDATION_AFTER_EXPIRY = 2;

	public StandardItem(String name, int sellIn, int quality){
		super(name, sellIn, quality);
	}

	@Override
	public void updateQuality(){
		if(isExpired()){
			decreaseQuality(DAILY_QUALITY_DEGREDATION_AFTER_EXPIRY);
		}else{
			decreaseQuality(DAILY_QUALITY_DEGREDATION_BEFORE_EXPIRY);
		}
	}
}
